tests/foreman/api/test_organization.py

```diff
@@ -266,3 +266,37 @@ def test_negative_update(self, attrs):
         )
         with self.assertRaises(HTTPError):
             response.raise_for_status()
+
+    def test_bugzilla_1103157(self):
+        """@Test: Create organization and add three compute resources
+        one by one using different transactions to see that they
+        actually added, but not overwrite each other
+
+        @Feature: Organization - Update
+
+        @Steps:
+
+        1. Create an organization
+        2. Create three compute resources
+        3. Update organization with three created compute resources
+
+        @Assert: Organization contains all three compute resources
+
+        """
+        for _ in range(3):
+            compute_resource_id = entities.ComputeResource(
+                name=gen_string('alpha'),
+                provider='Libvirt',
+                url='qemu://host.example.com/system'
+            ).create_json()['id']
+
+            client.put(
+                self.organization.path(),
+                {'organization': {
+                    'compute_resource_ids': compute_resource_id}},
+                verify=False,
+                auth=get_server_credentials(),
+            ).raise_for_status()
+
+        org_res = self.organization.read_json()
+        self.assertEqual(len(org_res['compute_resources']), 3)
```

Comment:
This would be more idiomatic and avoid single-use variables:

```python
self.assertEqual(
    3,
    len(self.organization.read_json()['compute_resources']),
)
```